.gitlab-ci.yml

```diff
@@ -1,14 +1,108 @@
-test:
-  image: "python:3.6-stretch"
-  script:
-    - apt-get update -qq
-    - apt-get install -y git
-    - apt-get install -y xsltproc
-    - apt-get install -y genisoimage
-    - apt-get install -y enchant
-    - apt-get install -y shellcheck
-    - apt-get install -y python3-pip
-    - pip3 install --upgrade pip
-    - pip install tox
+stages:
+  - test
+  - doc
+  - package
+  - deploy
+
+.fedora_install_deps: &fedora_install_deps
+  before_script:
+    - dnf install -y --refresh python3 python3-devel python3-pip
+    # requirements
+    - dnf install -y python3-docopt python3-lxml python3-pyyaml python3-pyxattr python3-six python3-future python3-requests
```

Comment:
Fedora has generic `python3dist(modname)` Provides for all Python modules, in case you might prefer to use those here: https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#_automatic_provides_with_a_standardized_name

Reply:
I have changed these accordingly.